specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json

```diff
@@ -0,0 +1,1661 @@
+        }
+      ],
+      "required": [
+        "location"
+      ]
+    },
+    "ServerProperties": {
+      "properties": {
+        "administratorLogin": {
+          "type": "string",
+          "description": "The administrator's login name of a server. Can only be specified when the server is being created (and is required for creation)."
+        },
+        "version": {
+          "$ref": "#/definitions/ServerVersion",
+          "description": "Server version."
+        },
+        "sslEnforcement": {
+          "$ref": "#/definitions/SslEnforcement",
+          "description": "Enable ssl enforcement or not when connect to server."
+        },
+        "userVisibleState": {
+          "type": "string",
+          "description": "A state of a server that is visible to user.",
+          "enum": [
+            "Ready",
+            "Dropping",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "name": "ServerState",
+            "modelAsString": true
+          }
+        },
+        "fullyQualifiedDomainName": {
+          "type": "string",
+          "description": "The fully qualified domain name of a server."
+        },
+        "EarliestRestoreDate": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Earliest restore point creation time (ISO8601 format)"
+        },
+        "storageProfile": {
+          "$ref": "#/definitions/StorageProfile",
+          "description": "Storage profile of a server."
+        }
+      },
+      "description": "The properties of a server."
+    },
+    "StorageProfile": {
+      "properties": {
+        "backupRetentionDays": {
+          "type": "integer",
+          "description": "Backup retention days for the server."
+        },
+        "geoRedundantBackup": {
+          "type": "string",
+          "description": "The tier of the particular SKU, e.g. Basic.",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "x-ms-enum": {
+            "name": "GeoRedundantBackup",
+            "modelAsString": true
+          }
+        },
+        "storageMB": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Max storage allowed for a server."
+        }
+    },
+    "description": "Storage Profile properties of a server"
+  },
+    "ServerPropertiesForCreate": {
+      "discriminator": "createMode",
+      "required": [ "createMode" ],
+      "properties": {       
+        "version": {
+          "$ref": "#/definitions/ServerVersion",
+          "description": "Server version."
+        },
+        "sslEnforcement": {
+          "$ref": "#/definitions/SslEnforcement",
+          "description": "Enable ssl enforcement or not when connect to server."
+        },
+        "storageProfile": {
+          "$ref": "#/definitions/StorageProfile",
+          "description": "Storage profile of a server."
+        },
+        "sku": {
```

Comment:
Remove sku from properties, please understand the request structure firstly.
{  
   "Location":"centralus",
   "Properties":{  
     "storageProfile": {
       ...
     },
     ...
   },
   "Sku":{
     ...
   },
   "Tags":{  
      ...
   }
}

Reply:
Thanks for catching this! I was only took from Reva's pull request and removed the VNET part. Seems this was missed from his. Removed this.